src/components/App/App.js

```diff
@@ -52,32 +82,60 @@ class App extends React.Component {
 					loadModel
 					confirmMatch={this.setPrediction}
 				/>
-				{this.props.prediction && (
-					<ConfirmationBox
-						item={this.props.prediction.index}
-						onYes={() => {
-							let id = getUserSlackID(
-								this.props.currentUser,
-								this.props.users
-							);
-							const name = labels[this.props.prediction.index][0];
-							sendSlackMessage(id, name, this.getStoreCode(name));
-							this.props.setPrediction(null);
-						}}
-						onNo={() => {
-							this.props.setPrediction(null);
-							this.props.setShowList(true);
-						}}>
-						<img src={this.props.prediction.img} alt="" />
-					</ConfirmationBox>
-				)}
-				{this.props.showList && <StoreListContainer />}
-				{this.props.slackUserFetchError && (
-					<ErrorMessage text="failed to fetch users" />
+				{this.props.prediction &&
+					this.props.storeList[this.props.prediction.id] && (
+						<ConfirmationBox
+							item={
+								this.props.storeList[this.props.prediction.id]
+							}
+							onYes={() => {
+								let id = getUserSlackID(
```

Comment:
Not introduced in this PR but it'd be good to extract both `onYes` and `onNo` handlers to class methods, this would allow us to keep render quite minimal and have the logic elsewhere

e.g.

```
<ConfirmationBox
  onYes={this.handleYes}
  onNo={this.handleNo}
/>

// ...

onYes = () => { ... }
onNo = () => { ... }
```